zerver/lib/display_recipient.py

```diff
@@ -122,101 +122,39 @@ def bulk_fetch_user_display_recipients(
     Returns dict mapping recipient_id to corresponding display_recipient
     """
 
-    if len(recipient_tuples) == 0:
-        return {}
+    get_recipient_id = lambda tup: tup[0]
+    get_type = lambda tup: tup[1]
 
-    recipient_id_to_type = {recipient[0]: recipient[1] for recipient in recipient_tuples}
+    personal_tuples = [tup for tup in recipient_tuples if get_type(tup) == Recipient.PERSONAL]
+    huddle_tuples = [tup for tup in recipient_tuples if get_type(tup) == Recipient.HUDDLE]
 
-    recipient_id_to_type_id = {recipient[0]: recipient[2] for recipient in recipient_tuples}
+    huddle_recipient_ids = [get_recipient_id(tup) for tup in huddle_tuples]
+    huddle_user_id_dict = bulk_get_huddle_user_ids(huddle_recipient_ids)
```

Comment:
Would `huddle_recipient_id_to_user_ids` be a clearer name for this?  `huddle_user_id_dict` doesn't really hint the types of keys/values.

Reply:
Fixing.